src/Tests/IQsharpEngineTests.cs

```diff
@@ -245,6 +278,50 @@ public async Task Estimate()
             await AssertEstimate(engine, "HelloQ");
         }
 
+        [TestMethod]
+        [TestCategory("Experimental")]
+        public async Task NoisySimulateWithTwoQubitOperation()
+        {
+            var engine = await Init();
+            var channel = new MockChannel();
+
+            // Compile it:
+            await AssertCompile(engine, SNIPPETS.SimpleDebugOperation, "SimpleDebugOperation");
+
+            // Try running again:
+            // Note that noiseModel: null sets the noise model to be ideal.
+            await AssertNoisySimulate(engine, "SimpleDebugOperation", noiseModel: null);
+        }
+
+        [TestMethod]
+        [TestCategory("Experimental")]
+        public async Task NoisySimulateWithFailIfOne()
+        {
+            var engine = await Init();
+            var channel = new MockChannel();
+
+            // Compile it:
+            await AssertCompile(engine, SNIPPETS.FailIfOne, "FailIfOne");
+
+            // Try running again:
+            // Note that noiseModel: null sets the noise model to be ideal.
+            await AssertNoisySimulate(engine, "FailIfOne", noiseModel: null);
```

Comment:
This might be a bit silly, but is it worth having the inverse test where a noise model is specified that is so noisy that we expect `FailIfOne` to fail? Perhaps that's too probabilistic of a test such that it would not be a good candidate for CI, but is there any way the noise model could be set such that freshly allocated qubits in the ground state have a 100% chance to measure a One?

Reply:
I think that could be a good test, but it'd be difficult to write for exactly the reason you mention: such a test would be inherently probabilistic. Given that probabilistic tests have been a problem for our build already, I'm reticent to add more, but I think the question you raise is a good one to address as a part of improving test coverage more generally. Perhaps something like a completely amplitude damping channel?